apis/python/src/tiledbsc/soma.py

```diff
@@ -64,6 +64,20 @@ def __init__(
         :param uri: URI of the TileDB group
         """
 
+        # People can (and should) call by name. However, it's easy to forget. For example,
```

Comment:
Found during debug. `soma = tiledbsc.SOMA(uri, ctx)` is incorrect while `soma = tiledbsc.SOMA(uri, ctx=ctx)` is correct. This needs to be surfaced proactively to the user.

Reply:
You may want to enforce [keyword-only arguments](https://peps.python.org/pep-3102/) to avoid such incorrect usage. 

Manual runtime type checking is considered unidiomatic in general and doesn't scale well with the complexity of annotations (think of `Dict[str, List[Optional[int]]`). For the cases that really need it, there are 3rd party libraries such as [Typeguard](https://typeguard.readthedocs.io/en/latest/userguide.html) or [Pydantic](https://pydantic-docs.helpmanual.io/) that use and enforce annotations at runtime transparently.